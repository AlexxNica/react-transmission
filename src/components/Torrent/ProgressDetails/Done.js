import React from 'react';
import { FormattedMessage, defineMessages, injectIntl } from 'react-intl';

import { size as formatSize, ratioString } from 'util/formatters';

const messages = defineMessages({
  uploadStats: {
    id: 'torrent.progress.done.uploadstats',
    defaultMessage: `uploaded {uploaded} (Ratio: {ratio})`,
  },
});

const PartialDone = injectIntl(({ torrent, intl }) =>
  <FormattedMessage
    id='torrent.progress.done.partial'
    defaultMessage={`{remaining} of {total} ({completed}), {stats}`}
    values={{
      remaining: formatSize(torrent.sizeWhenDone),
      total: formatSize(torrent.totalSize),
      completed: intl.formatNumber(torrent.percentDone, {style: 'percent'}),
      stats: intl.formatMessage(messages.uploadStats, {
        uploaded: formatSize(torrent.uploadedEver),
        ratio: ratioString(torrent.uploadRatio),
      }),
    }}
  />
);

const DoneDone = injectIntl(({ torrent, intl }) =>
  <FormattedMessage
    id='torrent.progress.done.done'
    defaultMessage={`{total}, {stats}`}
    values={{
      total: formatSize(torrent.totalSize),
      stats: intl.formatMessage(messages.uploadStats, {
        uploaded: formatSize(torrent.uploadedEver),
        ratio: ratioString(torrent.uploadRatio),
      }),
    }}
  />
);

function ProgressDone({ torrent }) {
  if (torrent.totalSize === torrent.sizeWhenDone) {
    return <DoneDone torrent={torrent} />;
  }

  return <PartialDone torrent={torrent} />;
}

export default ProgressDone;
