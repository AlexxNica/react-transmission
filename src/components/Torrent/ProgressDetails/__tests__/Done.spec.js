import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import Done from '../Done';

test('Done', () => {
  const component = createComponentWithIntl(
    <Done torrent={{sizeWhenDone: 1E5, totalSize: 1E5, percentDone: 1, uploadedEver: 1E5, uploadRatio: 2}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Done partially', () => {
  const component = createComponentWithIntl(
    <Done torrent={{sizeWhenDone: 1E4, totalSize: 1E5, percentDone: 0.1, uploadedEver: 1E4, uploadRatio: 2}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
