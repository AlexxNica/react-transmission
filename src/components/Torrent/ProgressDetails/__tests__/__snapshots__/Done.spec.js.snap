exports[`test Done 1`] = `
<span>
  100.0 kB, uploaded 100.0 kB (Ratio: 2.00)
</span>
`;

exports[`test Done partially 1`] = `
<span>
  10.0 kB of 100.0 kB (10%), uploaded 10.0 kB (Ratio: 2.00)
</span>
`;
